src/covers/discogscoverprovider.cpp

```diff
@@ -0,0 +1,373 @@
+    args.append(Arg("artist", s_ctx->artist.toLower()));
+  }
+  if (!s_ctx->album.isEmpty()) {
+    args.append(Arg("release_title", s_ctx->album.toLower()));
+  }
+
+  QUrlQuery url_query;
+  QUrl url(kUrlSearch);
+  QStringList query_items;
+
+  // Encode the arguments
+  for (const Arg& arg : args) {
+    EncodedArg encoded_arg(QUrl::toPercentEncoding(arg.first),
+                           QUrl::toPercentEncoding(arg.second));
+    query_items << QString(encoded_arg.first + "=" + encoded_arg.second);
+    url_query.addQueryItem(encoded_arg.first, encoded_arg.second);
+  }
+
+  // Sign the request
+  const QByteArray data_to_sign =
+      QString("GET\n%1\n%2\n%3")
+          .arg(url.host(), url.path(), query_items.join("&"))
+          .toLatin1();
+  const QByteArray signature(Utilities::HmacSha256(
+      QByteArray::fromBase64(kSecretKeyB64), data_to_sign));
+
+  // Add the signature to the request
+  url_query.addQueryItem("Signature",
+                         QUrl::toPercentEncoding(signature.toBase64()));
+
+  url.setQuery(url_query);
+  QNetworkReply* reply = network_->get(QNetworkRequest(url));
+
+  NewClosure(reply, SIGNAL(error(QNetworkReply::NetworkError)), this,
+             SLOT(SearchRequestError(QNetworkReply::NetworkError,
+                                     QNetworkReply*, int)),
+             reply, s_ctx->id);
+  NewClosure(reply, SIGNAL(finished()), this,
+             SLOT(HandleSearchReply(QNetworkReply*, int)), reply, s_ctx->id);
+}
+
+void DiscogsCoverProvider::SendReleaseRequest(
+    DiscogsCoverSearchContext* s_ctx, DiscogsCoverReleaseContext* r_ctx) {
+  typedef QPair<QString, QString> Arg;
+  typedef QList<Arg> ArgList;
+
+  typedef QPair<QByteArray, QByteArray> EncodedArg;
+  typedef QList<EncodedArg> EncodedArgList;
+
+  QUrlQuery url_query;
+  QStringList query_items;
+
+  ArgList args =
+      ArgList() << Arg("key", QByteArray::fromBase64(kAccessKeyB64))
+                << Arg("secret", QByteArray::fromBase64(kSecretKeyB64));
+  // Encode the arguments
+  for (const Arg& arg : args) {
+    EncodedArg encoded_arg(QUrl::toPercentEncoding(arg.first),
+                           QUrl::toPercentEncoding(arg.second));
+    query_items << QString(encoded_arg.first + "=" + encoded_arg.second);
+    url_query.addQueryItem(encoded_arg.first, encoded_arg.second);
+  }
+
+  QUrl url(r_ctx->resource_url);
+
+  // Sign the request
+  const QByteArray data_to_sign =
+      QString("GET\n%1\n%2\n%3")
+          .arg(url.host(), url.path(), query_items.join("&"))
+          .toLatin1();
+  const QByteArray signature(Utilities::HmacSha256(
+      QByteArray::fromBase64(kSecretKeyB64), data_to_sign));
+
+  // Add the signature to the request
+  url_query.addQueryItem("Signature",
+                         QUrl::toPercentEncoding(signature.toBase64()));
+
+  url.setQuery(url_query);
+  QNetworkReply* reply = network_->get(QNetworkRequest(url));
+
+  NewClosure(reply, SIGNAL(error(QNetworkReply::NetworkError)), this,
```

Comment:
Pretty sure signal arguments don't work with `NewClosure`

Reply:
Won't work without, throws errors. It works like this. it's already done like this in other parts of the code like gpodder.